tests/integration/s3_asset_test.py

```diff
@@ -16,6 +25,42 @@
 ARN = "arn:aws:s3:::"
 BUCKET_NAME = MODULE_NAME
 BUCKET_ARN = f"{ARN}{MODULE_NAME}"
+OBJECT_NAME = f"myobject_{MODULE_NAME}.csv"
+OBJECT_ARN = f"{ARN}{BUCKET_NAME}/prefix/{OBJECT_NAME}"
+CERTIFICATE_STATUS = CertificateStatus.VERIFIED
+CERTIFICATE_MESSAGE = "Automated testing of the Python SDK."
+ANNOUNCEMENT_TYPE = AnnouncementType.INFORMATION
+ANNOUNCEMENT_TITLE = "Python SDK testing."
+ANNOUNCEMENT_MESSAGE = "Automated testing of the Python SDK."
+
+
+def block(
+    client: AtlanClient, response: AssetMutationResponse
+) -> AssetMutationResponse:
+    if response.mutated_entities and response.mutated_entities.DELETE:
+        _retrieve_and_check(client, response.mutated_entities.DELETE, 0)
+    return response
+
+
+def _retrieve_and_check(client: AtlanClient, to_check: list[Asset], retry_count: int):
+    leftovers = []
+    for one in to_check:
+        try:
+            candidate = client.get_asset_by_guid(one.guid, asset_type=type(one))
+            if candidate and candidate.status == EntityStatus.ACTIVE:
+                leftovers.append(candidate)
+        except NotFoundError:
+            # If it is not found, it was successfully deleted (purged), so we
+            # do not need to look for it any further
+            print("Asset no longer exists.")
+        except AtlanError:
+            leftovers.append(one)
+    if leftovers:
+        if retry_count == 20:
+            raise AtlanError(message="Overran retry limit", code="500", status_code=500)
+        else:
```

Comment:
This else isn't needed since an exception is being raised.

Reply:
The if retry_count == 20 is needed to check if the exception should be raised. The else isn't needed because when the exception is raised the code block will be exited.

Reply:
Ooohhh, I see what you mean now 😁